workshops/models.py

```diff
@@ -97,11 +98,13 @@ class Person(models.Model):
     github     = models.CharField(max_length=STR_MED, unique=True, null=True)
     twitter    = models.CharField(max_length=STR_MED, unique=True, null=True)
     url        = models.CharField(max_length=STR_LONG, null=True)
-    slug       = models.CharField(max_length=STR_LONG, null=True)
+    username   = models.CharField(max_length=STR_MED, unique=True)
 
     # Set the custom manager
     objects = PersonManager()
 
+    USERNAME_FIELD = 'username'
```

Comment:
On Sun, Jan 18, 2015 at 08:37:45AM -0800, Piotr Banaszkiewicz wrote:
> > +    USERNAME_FIELD = 'username'
> 
> I might have missed that in all these lengthy discussions about
> auth, but why add a new field? We can use `email`, as it's already a
> unique field.

For the same reason we can't use the slug: the username field needs to
be unique, and Django doesn't do a good job with unique string fields
that aren't required [1].  If we required email addresses for every
Person, using email addresses would be fine, but I don't think we can
do that.

[1]: https://github.com/swcarpentry/amy/pull/125#issuecomment-70414088